src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/DeserializationHelpers.cs

```diff
@@ -57,6 +58,37 @@ internal static class DeserializationHelpers
             }
         }
 
+
+        internal static void ApplyInstanceAnnotations(object resource, IEdmStructuredTypeReference structuredType, ODataResourceWrapper resourceWrapper, 
+            ODataDeserializerProvider deserializerProvider, ODataDeserializerContext readContext)
+        {
+            IDelta delta = resource as IDelta;
+            
+            PropertyInfo propertyInfo = EdmLibHelpers.GetInstanceAnnotationsDictionary(structuredType.StructuredDefinition(), readContext.Model);
+            if (propertyInfo == null)
+            {
+                return;
+            }
+
+            object value;
+            if (delta != null)
+            {                
+                delta.TryGetPropertyValue(propertyInfo.Name, out value);
+            }
+            else
+            {
+                value = propertyInfo.GetValue(resource);
+            }
+
+            Dictionary<string, IDictionary<string, object>> instAnnotationDictionary = value as Dictionary<string, IDictionary<string, object>>;
```

Comment:
>Dictionary [](start = 96, length = 10)

Shouldn't this be IDictionary<string,IDictionary<string,object>> rather than *Dictionary*<string,IDictionary<string,object>>?

Reply:
#Resolved